### DISCUSSION TITLE
Add support associations for each product variants too

### DISCUSSION THREAD

**Discussion by devmansurov** [lunarphp/lunar#74] [Feature Requests]
Now we can associate products with each other with products. But it would be great if we could associate products for each variant separately too. Or leave it as it is now, but add the ability for associated products in which product variants it is available and in which it is not available.

**Comment by glennjacobs** [comment]
Can you provide me with a use case?

**Reply by devmansurov** [reply]
For example, I sell cakes. I will also sell decorations for this cake. These decorations may only be available in certain cake weight categories. And each cake weight is a separate version of the product. Therefore, I want to add these decorations only to certain product variants.

Second example. I sell phones and headphones. They have different colors. Based on the color, I want to associate some products that match the selected color from the user's side when buying a product. Each product color is a separate product variant.

Example:

```
$variant->associate(
    \GetCandy\Models\ProductVariant $crossSellProduct,
    'custom-type'
);
```
